service/storage_service.cc

```diff
@@ -2544,15 +2552,83 @@ class topology_coordinator {
         , _raft(raft_server), _term(raft_server.get_current_term())
         , _raft_topology_cmd_handler(std::move(raft_topology_cmd_handler))
         , _tablet_allocator(tablet_allocator)
+        , _load_stats_refresh_trigger([this] {
+            // Awakes the topology event handler, which allows stats to be refreshed
+            // if coordinator allows for handling of tablet migrations.
+            _topo_sm.event.signal();
+        })
         , _ring_delay(ring_delay)
     {}
 
     future<> run();
 };
 
+future<locator::load_stats> topology_coordinator::refresh_table_load_stats() {
+    locator::load_stats stats;
+
+    auto tm = get_token_metadata_ptr();
+    auto& topology = tm->get_topology();
+
+    for (auto& [dc, endpoints] : topology.get_datacenter_endpoints()) {
+        locator::load_stats dc_stats;
+        slogger.info("raft topology: Refreshing table load stats for DC {} that has {} endpoints", dc, endpoints.size());
+        for (auto& endpoint : endpoints) {
```

Comment:
It's better to iterate on hosts and translate to endpoints only when making the rpc call. That's the direction the entire system is moving.

Reply:
Ok.